<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Sibei Liu" />


<title>Mendelian randomization for pleiotropic effects using GWAS summary</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/anchor-sections-1.0/anchor-sections.css" rel="stylesheet" />
<script src="site_libs/anchor-sections-1.0/anchor-sections.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">bivariate-models</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
<li>
  <a href="MR-Sibei.html">MR for pleiotropic effects</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/cumc/bivariate-models">Git</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Mendelian randomization for pleiotropic effects using GWAS summary</h1>
<h4 class="author">Sibei Liu</h4>
<h4 class="date">2021/2/1</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-02-08
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>bivariate-models/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210203code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210203)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210203code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210203)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomSibeiliubivariatemodelstree1b2d7bbc22d52397ee17984dde4da3cf001b31fftargetblank1b2d7bba"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/Sibeiliu/bivariate-models/tree/1b2d7bbc22d52397ee17984dde4da3cf001b31ff" target="_blank">1b2d7bb</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomSibeiliubivariatemodelstree1b2d7bbc22d52397ee17984dde4da3cf001b31fftargetblank1b2d7bba" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/Sibeiliu/bivariate-models/tree/1b2d7bbc22d52397ee17984dde4da3cf001b31ff" target="_blank">1b2d7bb</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Unstaged changes:
    Modified:   analysis/MR-Sibei.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/MR-Sibei.Rmd</code>) and HTML (<code>docs/MR-Sibei.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Sibeiliu/bivariate-models/blob/1b2d7bbc22d52397ee17984dde4da3cf001b31ff/analysis/MR-Sibei.Rmd" target="_blank">1b2d7bb</a>
</td>
<td>
Sibei Liu
</td>
<td>
2021-02-08
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Sibeiliu/bivariate-models/1b2d7bbc22d52397ee17984dde4da3cf001b31ff/docs/MR-Sibei.html" target="_blank">1b2d7bb</a>
</td>
<td>
Sibei Liu
</td>
<td>
2021-02-08
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Sibeiliu/bivariate-models/blob/bdcc2fd9cc6f1a00838b24504fddb2b3edd38fdc/analysis/MR-Sibei.Rmd" target="_blank">bdcc2fd</a>
</td>
<td>
Sibei Liu
</td>
<td>
2021-02-08
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Sibeiliu/bivariate-models/bdcc2fd9cc6f1a00838b24504fddb2b3edd38fdc/docs/MR-Sibei.html" target="_blank">bdcc2fd</a>
</td>
<td>
Sibei Liu
</td>
<td>
2021-02-08
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Sibeiliu/bivariate-models/blob/3920d1137bc6cd61f66960f1c1da9b995fd97d6c/analysis/MR-Sibei.Rmd" target="_blank">3920d11</a>
</td>
<td>
Sibei Liu
</td>
<td>
2021-02-07
</td>
<td>
add code
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Sibeiliu/bivariate-models/3920d1137bc6cd61f66960f1c1da9b995fd97d6c/docs/MR-Sibei.html" target="_blank">3920d11</a>
</td>
<td>
Sibei Liu
</td>
<td>
2021-02-07
</td>
<td>
add code
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Sibeiliu/bivariate-models/blob/3ce71b3fc85b5bf17b9bb4040eeabe98ede4836f/analysis/MR-Sibei.Rmd" target="_blank">3ce71b3</a>
</td>
<td>
Sibei Liu
</td>
<td>
2021-02-05
</td>
<td>
format edit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Sibeiliu/bivariate-models/3ce71b3fc85b5bf17b9bb4040eeabe98ede4836f/docs/MR-Sibei.html" target="_blank">3ce71b3</a>
</td>
<td>
Sibei Liu
</td>
<td>
2021-02-05
</td>
<td>
format edit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Sibeiliu/bivariate-models/blob/4a4469210f169c355adcdfdb029fd1baaf485fa0/analysis/MR-Sibei.Rmd" target="_blank">4a44692</a>
</td>
<td>
Sibei Liu
</td>
<td>
2021-02-05
</td>
<td>
transform the data
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Sibeiliu/bivariate-models/4a4469210f169c355adcdfdb029fd1baaf485fa0/docs/MR-Sibei.html" target="_blank">4a44692</a>
</td>
<td>
Sibei Liu
</td>
<td>
2021-02-05
</td>
<td>
transform the data
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Sibeiliu/bivariate-models/blob/8cff0788140c6ebb3678077519407d53cabcc763/analysis/MR-Sibei.Rmd" target="_blank">8cff078</a>
</td>
<td>
Sibei Liu
</td>
<td>
2021-02-05
</td>
<td>
package installed
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Sibeiliu/bivariate-models/8cff0788140c6ebb3678077519407d53cabcc763/docs/MR-Sibei.html" target="_blank">8cff078</a>
</td>
<td>
Sibei Liu
</td>
<td>
2021-02-05
</td>
<td>
package installed
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Sibeiliu/bivariate-models/blob/c698cab481a2db8e7c748b25cd70861f727d8586/analysis/MR-Sibei.Rmd" target="_blank">c698cab</a>
</td>
<td>
Sibei Liu
</td>
<td>
2021-02-03
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Sibeiliu/bivariate-models/c698cab481a2db8e7c748b25cd70861f727d8586/docs/MR-Sibei.html" target="_blank">c698cab</a>
</td>
<td>
Sibei Liu
</td>
<td>
2021-02-03
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Sibeiliu/bivariate-models/blob/d9e1b1c5f4134ade1e035bad75edff26ca964eb4/analysis/MR-Sibei.Rmd" target="_blank">d9e1b1c</a>
</td>
<td>
Sibei Liu
</td>
<td>
2021-02-03
</td>
<td>
web updtae
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Sibeiliu/bivariate-models/d9e1b1c5f4134ade1e035bad75edff26ca964eb4/docs/MR-Sibei.html" target="_blank">d9e1b1c</a>
</td>
<td>
Sibei Liu
</td>
<td>
2021-02-03
</td>
<td>
web updtae
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Sibeiliu/bivariate-models/blob/d38225c5eae39e8c95560da646fef6b593837fe8/analysis/MR-Sibei.Rmd" target="_blank">d38225c</a>
</td>
<td>
Sibei Liu
</td>
<td>
2021-02-03
</td>
<td>
add some files
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Sibeiliu/bivariate-models/d38225c5eae39e8c95560da646fef6b593837fe8/docs/MR-Sibei.html" target="_blank">d38225c</a>
</td>
<td>
Sibei Liu
</td>
<td>
2021-02-03
</td>
<td>
add some files
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre><code>          asthma_V1 asthma_CHR asthma_POS asthma_REF asthma_ALT
1:  16_11031741_A_G         16   11031741          G          A
2:  16_11031846_C_T         16   11031846          T          C
3:  16_11031883_A_C         16   11031883          A          C
4:  16_11032195_A_G         16   11032195          A          G
5: 16_11032226_G_GA         16   11032226          G         GA
6:  16_11032424_A_C         16   11032424          A          C
         asthma_SNP  asthma_BETA  asthma_SE    asthma_P    asthma_Z
1:       rs79748582  0.008690250 0.00105534 1.80204e-16  8.23456981
2:      rs575622542 -0.011790400 0.00721980 1.02456e-01 -1.63306180
3:      rs560893521  0.004589740 0.01021210 6.53115e-01  0.44943900
4:       rs74245513 -0.002022650 0.00134911 1.33809e-01 -1.49924918
5: 16:11032226_GA_G -0.000110200 0.00119135 9.26300e-01 -0.09250100
6:       rs79429692  0.000252771 0.00767281 9.73719e-01  0.03294431</code></pre>
<pre><code>             BMI_V1 BMI_CHR   BMI_BP BMI_REF BMI_ALT          BMI_SNP
1:      rs575622542      16 11031846       T       C      rs575622542
2:      rs560893521      16 11031883       A       C      rs560893521
3:       rs74245513      16 11032195       A       G       rs74245513
4: 16:11032226_GA_G      16 11032226       G      GA 16:11032226_GA_G
5:       rs79429692      16 11032424       A       C       rs79429692
6:       rs76284787      16 11032774       A       G       rs76284787
      BMI_BETA     BMI_SE BMI_P      BMI_Z
1:  0.01423150 0.01837130 0.570  0.5680515
2:  0.05144910 0.02585070 0.058  1.8956979
3:  0.00216789 0.00343834 0.510  0.6588377
4:  0.00375400 0.00303706 0.220  1.2265281
5: -0.00591978 0.01938120 0.810 -0.2404260
6:  0.00933315 0.01020530 0.350  0.9345893</code></pre>
<div id="step1-prepare-the-data" class="section level1">
<h1>Step1: Prepare the data</h1>
<pre class="r"><code># for merge two data traits and finding the common SNPs
data &lt;- gwas_merge(asthma, BMI, snp_name_cols = c(&quot;asthma_SNP&quot;, &quot;BMI_SNP&quot;), 
                       beta_hat_cols = c(&quot;asthma_BETA&quot;, &quot;BMI_BETA&quot;), 
                       se_cols = c(&quot;asthma_SE&quot;, &quot;BMI_SE&quot;), 
                       A1_cols = c(&quot;asthma_REF&quot;, &quot;BMI_REF&quot;), 
                       A2_cols = c(&quot;asthma_ALT&quot;, &quot;BMI_ALT&quot;))</code></pre>
<pre><code>Formatting X1
There are  6975  variants.
Removing  3529  duplicated variants leaving  1314 variants.
Removing  51  variants with illegal allelse leaving  1314 variants.
Removed  184  variants with ambiguous strand.
Flipping strand and effect allele so A1 is always A
Returning  1130  variants.
Formatting X2
There are  8862  variants.
Removing  4590  duplicated variants leaving  1560 variants.
Removing  57  variants with illegal allelse leaving  1560 variants.
Removed  230  variants with ambiguous strand.
Flipping strand and effect allele so A1 is always A
Returning  1330  variants.
After merging and removing variants with inconsistent alleles,  there are  1081  variants that are present in both studies and can be used with CAUSE.</code></pre>
<pre class="r"><code># formated data
head(data)</code></pre>
<pre><code>          snp   beta_hat_1        seb1   beta_hat_2       seb2 A1 A2
1 rs150258910 -4.42786e-03 0.003055580 -0.000894756 0.00778383  A  C
2  rs12922090 -6.20602e-04 0.001100210 -0.000689250 0.00280498  A  G
3  rs12922409 -6.05009e-04 0.001100350 -0.000729873 0.00280532  A  G
4 rs539184488 -5.97138e-05 0.010583600  0.007837880 0.02683000  A  G
5  rs35051965  8.68803e-04 0.000944880 -0.000741283 0.00240700  A  G
6  rs56136462  1.79333e-03 0.000962815  0.002083740 0.00245190  A  G</code></pre>
<pre class="r"><code>dim(data)</code></pre>
<pre><code>[1] 1081    7</code></pre>
<p>Warning: Among the two traits, only 1081 common SNP variants, small number will lead to poor estimates of the mixing parameters whih leads to bad model comparisons</p>
</div>
<div id="step-2-analysis-calculate-the-nuisance-parameter" class="section level1">
<h1>Step 2: Analysis-calculate the nuisance parameter</h1>
<pre class="r"><code>set.seed(100)
varlist &lt;- data$snp
params &lt;- est_cause_params(data, varlist)</code></pre>
<pre><code>Estimating CAUSE parameters with  1081  variants.
1 0.3226372 
2 0.008047968 
3 0.0003166234 
4 1.258583e-05 
5 5.004749e-07 
6 1.985126e-08 </code></pre>
<pre class="r"><code>params</code></pre>
<pre><code>$rho
[1] 0.3310149

$pi
 [1] 0.16204943 0.00000000 0.48408372 0.10515948 0.00000000 0.00000000
 [7] 0.08757081 0.16113657 0.00000000 0.00000000 0.00000000 0.00000000
[13] 0.00000000 0.00000000 0.00000000 0.00000000 0.00000000 0.00000000
[19] 0.00000000 0.00000000 0.00000000 0.00000000 0.00000000 0.00000000

$mix_grid
           S1         S2         pi
1 0.000000000 0.00000000 0.16204943
2 0.004273606 0.00000000 0.48408372
3 0.017094425 0.00000000 0.10515948
4 0.000000000 0.04468453 0.08757081
5 0.003021896 0.04468453 0.16113657

$loglik
[1] 7255.658

$PIS
                         pi         pi         pi         pi         pi
 [1,] 0.14566468 0.16188921 0.16204346 0.16204919 0.16204942 0.16204943
 [2,] 0.00000000 0.00000000 0.00000000 0.00000000 0.00000000 0.00000000
 [3,] 0.50701234 0.48470790 0.48410828 0.48408470 0.48408376 0.48408372
 [4,] 0.10067776 0.10488999 0.10514870 0.10515905 0.10515946 0.10515948
 [5,] 0.00000000 0.00000000 0.00000000 0.00000000 0.00000000 0.00000000
 [6,] 0.00000000 0.00000000 0.00000000 0.00000000 0.00000000 0.00000000
 [7,] 0.07784049 0.08726707 0.08755877 0.08757033 0.08757079 0.08757081
 [8,] 0.16880474 0.16124584 0.16114079 0.16113673 0.16113657 0.16113657
 [9,] 0.00000000 0.00000000 0.00000000 0.00000000 0.00000000 0.00000000
[10,] 0.00000000 0.00000000 0.00000000 0.00000000 0.00000000 0.00000000
[11,] 0.00000000 0.00000000 0.00000000 0.00000000 0.00000000 0.00000000
[12,] 0.00000000 0.00000000 0.00000000 0.00000000 0.00000000 0.00000000
[13,] 0.00000000 0.00000000 0.00000000 0.00000000 0.00000000 0.00000000
[14,] 0.00000000 0.00000000 0.00000000 0.00000000 0.00000000 0.00000000
[15,] 0.00000000 0.00000000 0.00000000 0.00000000 0.00000000 0.00000000
[16,] 0.00000000 0.00000000 0.00000000 0.00000000 0.00000000 0.00000000
[17,] 0.00000000 0.00000000 0.00000000 0.00000000 0.00000000 0.00000000
[18,] 0.00000000 0.00000000 0.00000000 0.00000000 0.00000000 0.00000000
[19,] 0.00000000 0.00000000 0.00000000 0.00000000 0.00000000 0.00000000
[20,] 0.00000000 0.00000000 0.00000000 0.00000000 0.00000000 0.00000000
[21,] 0.00000000 0.00000000 0.00000000 0.00000000 0.00000000 0.00000000
[22,] 0.00000000 0.00000000 0.00000000 0.00000000 0.00000000 0.00000000
[23,] 0.00000000 0.00000000 0.00000000 0.00000000 0.00000000 0.00000000
[24,] 0.00000000 0.00000000 0.00000000 0.00000000 0.00000000 0.00000000
              pi
 [1,] 0.16204943
 [2,] 0.00000000
 [3,] 0.48408372
 [4,] 0.10515948
 [5,] 0.00000000
 [6,] 0.00000000
 [7,] 0.08757081
 [8,] 0.16113657
 [9,] 0.00000000
[10,] 0.00000000
[11,] 0.00000000
[12,] 0.00000000
[13,] 0.00000000
[14,] 0.00000000
[15,] 0.00000000
[16,] 0.00000000
[17,] 0.00000000
[18,] 0.00000000
[19,] 0.00000000
[20,] 0.00000000
[21,] 0.00000000
[22,] 0.00000000
[23,] 0.00000000
[24,] 0.00000000

$RHO
[1] 0.0000000 0.3226372 0.3306852 0.3310018 0.3310144 0.3310149 0.3310149

$LLS
 [1] 7244.325 7255.280 7255.650 7255.658 7255.658 7255.658 7255.658 7255.658
 [9] 7255.658 7255.658 7255.658 7255.658 7255.658

$converged
[1] TRUE

$prior
[1] 51.91503

$var
            [,1]
[1,] 0.003927044

attr(,&quot;class&quot;)
[1] &quot;cause_params&quot;</code></pre>
<pre class="r"><code>head(params$mix_grid)</code></pre>
<pre><code>           S1         S2         pi
1 0.000000000 0.00000000 0.16204943
2 0.004273606 0.00000000 0.48408372
3 0.017094425 0.00000000 0.10515948
4 0.000000000 0.04468453 0.08757081
5 0.003021896 0.04468453 0.16113657</code></pre>
<p>We have estimated that 16% of variants have trait 1 variance and trait 2 equal to 0 meaning that they have no association with either trait.</p>
</div>
<div id="step-3ld-pruning" class="section level1">
<h1>Step 3:LD Pruning</h1>
<p>This step requires LD estimates. Download estimates of chromsome 16 made in the 1000 Genomes CEU population from “<a href="https://zenodo.org/record/1464357#.W8a-fxROmV4" class="uri">https://zenodo.org/record/1464357#.W8a-fxROmV4</a>”</p>
<pre class="r"><code>#download.file(&quot;https://zenodo.org/record/1464357/files/chr16_AF0.05_0.1.RDS?download=1&quot;, destfile = &quot;../chr16_AF0.05_0.1.RDS&quot;)
#download.file(&quot;https://zenodo.org/record/1464357/files/chr16_AF0.05_snpdata.RDS?download=1&quot;, destfile=&quot;../chr16_AF0.05_snpdata.RDS&quot;)</code></pre>
<pre class="r"><code>ld &lt;- readRDS(&quot;../chr16_AF0.05_0.1.RDS&quot;)
snp_info &lt;- readRDS(&quot;../chr16_AF0.05_snpdata.RDS&quot;)

head(ld)</code></pre>
<pre><code>      rowsnp      colsnp        r2
1 rs76368850 rs542544747 0.2655372
2 rs76368850   rs4021615 0.1397717
3 rs76368850 rs201731689 0.1772501
4 rs76368850 rs572534626 0.1187799
5 rs76368850 rs565960587 0.1528682
6 rs76368850 rs558303720 0.1528682</code></pre>
<pre class="r"><code>head(snp_info)</code></pre>
<pre><code># A tibble: 6 x 9
     AF SNP         allele   chr   pos   snp_id region_id   map ld_snp_id
  &lt;dbl&gt; &lt;chr&gt;       &lt;chr&gt;  &lt;int&gt; &lt;int&gt;    &lt;int&gt;     &lt;int&gt; &lt;dbl&gt;     &lt;int&gt;
1 0.364 rs76368850  T,C       16 60291 67777509        16     0  67777509
2 0.439 rs542544747 A,C       16 60842 67777531        16     0  67777531
3 0.207 rs4021615   G,A       16 61349 67777546        16     0  67777546
4 0.581 rs201731689 C,G       16 61730 67777556        16     0  67777556
5 0.606 rs572534626 C,G       16 61977 67777564        16     0  67777564
6 0.838 rs565960587 C,A       16 62511 67777581        16     0  67777581</code></pre>
<p>The snp_info data frame contains information about all of the chromosome #16 variants with allele frequency greater than 0.05. The only piece of information we need from this data frame is the list of variants SNP column in snp_info data set which provides the total list of variants used in the LD calculations.</p>
<pre class="r"><code>variants &lt;- data %&gt;% mutate(pval1 = 2*pnorm(abs(beta_hat_1/seb1), lower.tail=FALSE))
pruned &lt;- ld_prune(variants = variants, 
                            ld = ld, total_ld_variants = snp_info$SNP, 
                            pval_cols = c(&quot;pval1&quot;), 
                            pval_thresh = c(1e-3))</code></pre>
<pre><code>You have suppplied information for  1081  variants.
Of these,  544  have LD information.</code></pre>
<pre class="r"><code>length(pruned)</code></pre>
<pre><code>[1] 6</code></pre>
</div>
<div id="step4-fit-cause" class="section level1">
<h1>Step4: Fit CAUSE</h1>
<pre class="r"><code>res &lt;- cause(X=data, variants = pruned, param_ests = params)</code></pre>
<pre><code>Estimating CAUSE posteriors using  6  variants.</code></pre>
</div>
<div id="step-5-results" class="section level1">
<h1>Step 5: Results</h1>
<pre class="r"><code>names(res)</code></pre>
<pre><code>[1] &quot;sharing&quot; &quot;causal&quot;  &quot;elpd&quot;    &quot;loos&quot;    &quot;data&quot;    &quot;sigma_g&quot; &quot;qalpha&quot; 
[8] &quot;qbeta&quot;  </code></pre>
<pre class="r"><code>res$elpd</code></pre>
<pre><code>   model1  model2 delta_elpd se_delta_elpd         z
1    null sharing 0.06539795    0.02429909 2.6913738
2    null  causal 0.18539268    0.25538647 0.7259299
3 sharing  causal 0.11999473    0.23675675 0.5068271</code></pre>
<p>model1, model2: The models being compared</p>
<p>delta_elpd: Estimated difference in elpd. If delta_elpd is negative, model 2 is a better fit</p>
<p>se_delta_elpd: Estimated standard error of delta_elpd</p>
<p>z: delta_elpd/se_delta_elpd. A z-score that can be compared to a normal distribution to test if the difference in model fit is significant.</p>
<pre class="r"><code>summary(res, ci_size=0.95)</code></pre>
<pre><code>p-value testing that causal model is a better fit:  0.69 
Posterior medians and  95 % credible intervals:
     model     gamma                eta                  q              
[1,] &quot;Sharing&quot; NA                   &quot;0.03 (-0.76, 0.78)&quot; &quot;0.06 (0, 0.3)&quot;
[2,] &quot;Causal&quot;  &quot;0.15 (-0.26, 0.53)&quot; &quot;0.01 (-0.77, 0.77)&quot; &quot;0.06 (0, 0.3)&quot;</code></pre>
<pre class="r"><code>plot(res$sharing)</code></pre>
<p><img src="figure/MR-Sibei.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot(res, type=&quot;data&quot;)</code></pre>
<p><img src="figure/MR-Sibei.Rmd/unnamed-chunk-11-2.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.3 (2020-10-10)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19042)

Matrix products: default

locale:
[1] LC_COLLATE=Chinese (Simplified)_China.936 
[2] LC_CTYPE=Chinese (Simplified)_China.936   
[3] LC_MONETARY=Chinese (Simplified)_China.936
[4] LC_NUMERIC=C                              
[5] LC_TIME=Chinese (Simplified)_China.936    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] installr_0.22.0   workflowr_1.6.2   eply_0.1.2        ieugwasr_0.1.5   
 [5] data.table_1.13.2 cause_1.0.0       forcats_0.5.0     stringr_1.4.0    
 [9] dplyr_1.0.2       purrr_0.3.4       tidyr_1.1.2       tibble_3.0.4     
[13] ggplot2_3.3.2     tidyverse_1.3.0   readr_1.4.0      

loaded via a namespace (and not attached):
 [1] matrixStats_0.57.0  fs_1.5.0            lubridate_1.7.9.2  
 [4] doParallel_1.0.16   httr_1.4.2          rprojroot_2.0.2    
 [7] numDeriv_2016.8-1.1 tools_4.0.3         backports_1.2.0    
[10] utf8_1.1.4          R6_2.5.0            irlba_2.3.3        
[13] DBI_1.1.0           colorspace_2.0-0    withr_2.3.0        
[16] tidyselect_1.1.0    gridExtra_2.3       compiler_4.0.3     
[19] git2r_0.28.0        cli_2.1.0           rvest_0.3.6        
[22] xml2_1.3.2          labeling_0.4.2      scales_1.1.1       
[25] SQUAREM_2021.1      mixsqp_0.3-43       digest_0.6.27      
[28] R.utils_2.10.1      rmarkdown_2.5       pscl_1.5.5         
[31] pkgconfig_2.0.3     htmltools_0.5.0     dbplyr_2.0.0       
[34] rlang_0.4.10        readxl_1.3.1        rstudioapi_0.13    
[37] farver_2.0.3        generics_0.1.0      jsonlite_1.7.2     
[40] R.oo_1.24.0         magrittr_2.0.1      loo_2.4.1          
[43] Matrix_1.2-18       Rcpp_1.0.6          munsell_0.5.0      
[46] fansi_0.4.1         R.methodsS3_1.8.1   lifecycle_0.2.0    
[49] stringi_1.5.3       whisker_0.4         yaml_2.2.1         
[52] MASS_7.3-53         grid_4.0.3          parallel_4.0.3     
[55] promises_1.1.1      crayon_1.3.4        lattice_0.20-41    
[58] haven_2.3.1         hms_0.5.3           knitr_1.30         
[61] pillar_1.4.6        codetools_0.2-16    reprex_0.3.0       
[64] glue_1.4.2          evaluate_0.14       RcppParallel_5.0.2 
[67] modelr_0.1.8        vctrs_0.3.6         httpuv_1.5.4       
[70] foreach_1.5.1       cellranger_1.1.0    gtable_0.3.0       
[73] assertthat_0.2.1    ashr_2.2-32         xfun_0.19          
[76] broom_0.7.2         later_1.1.0.1       truncnorm_1.0-8    
[79] intervals_0.15.2    iterators_1.0.13    ellipsis_0.3.1     </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
