---
title: "Mendelian randomization for pleiotropic effects using GWAS summary"
author: "Sibei Liu"
date: "2021/2/1"
output:
  html_document:
    code_folding: "hide"
---

```{r setup,include=F}
knitr::opts_chunk$set( message= FALSE,warning = FALSE)
library(readr)
library(tidyverse)
library(cause)
library(data.table)
library(ieugwasr)
library(eply)
library(workflowr)
library(installr)
```


```{r, echo = FALSE}
#try=fread("../ukb_imp_v3.Asthma_casesbyICD10codesANDselfreport_controlsbyselfreportandicd10_noautoimmuneincontrols.fastGWA.snp_stats.all_chr_16_53797565_53848561.genotype.gz")

#try2=fread("../ukb_imp_v3.Asthma_casesbyICD10codesANDselfreport_controlsbyselfreportandicd10_noautoimmuneincontrols.fastGWA.snp_stats.all_chr_16_53797565_53848561.population_ld.gz")
#######################################################
asthma_16_11031741_11307024=fread("../ukb_imp_v3.Asthma_casesbyICD10codesANDselfreport_controlsbyselfreportandicd10_noautoimmuneincontrols.fastGWA.snp_stats.all_chr_16_11031741_11307024.sumstats.gz")

asthma_16_11031741_11414898=fread("../ukb_imp_v3.Asthma_casesbyICD10codesANDselfreport_controlsbyselfreportandicd10_noautoimmuneincontrols.fastGWA.snp_stats.all_chr_16_11031741_11414898.sumstats.gz")

asthma_16_11034728_11450487=fread("../ukb_imp_v3.Asthma_casesbyICD10codesANDselfreport_controlsbyselfreportandicd10_noautoimmuneincontrols.fastGWA.snp_stats.all_chr_16_11034728_11450487.sumstats.gz")

asthma_16_27202702_27371397=fread("../ukb_imp_v3.Asthma_casesbyICD10codesANDselfreport_controlsbyselfreportandicd10_noautoimmuneincontrols.fastGWA.snp_stats.all_chr_16_27202702_27371397.sumstats.gz")

asthma_16_53797565_53848561=fread("../ukb_imp_v3.Asthma_casesbyICD10codesANDselfreport_controlsbyselfreportandicd10_noautoimmuneincontrols.fastGWA.snp_stats.all_chr_16_53797565_53848561.sumstats.gz")

asthma_colnames=colnames(asthma_16_53797565_53848561)
final_asthma_colnemas=str_c("asthma_",asthma_colnames)
asthma=rbind(asthma_16_11031741_11307024,asthma_16_11031741_11414898,asthma_16_11034728_11450487,asthma_16_27202702_27371397,asthma_16_53797565_53848561)

colnames(asthma)=final_asthma_colnemas
head(asthma)
```


```{r, echo = FALSE}
#################################
BMI_16_11031741_11307024=fread("../ukb_imp_v3.UKB_caucasians_BMIwaisthip_AsthmaAndT2D_INT-BMI_withagesex_041720.BoltLMM.snp_stats.all_chr_16_11031741_11307024.sumstats.gz")

BMI_16_11031741_11414898=fread("../ukb_imp_v3.UKB_caucasians_BMIwaisthip_AsthmaAndT2D_INT-BMI_withagesex_041720.BoltLMM.snp_stats.all_chr_16_11031741_11414898.sumstats.gz")

BMI_16_11034728_11450487=fread("../ukb_imp_v3.UKB_caucasians_BMIwaisthip_AsthmaAndT2D_INT-BMI_withagesex_041720.BoltLMM.snp_stats.all_chr_16_11034728_11450487.sumstats.gz")

BMI_16_27202702_27371397=fread("../ukb_imp_v3.UKB_caucasians_BMIwaisthip_AsthmaAndT2D_INT-BMI_withagesex_041720.BoltLMM.snp_stats.all_chr_16_27202702_27371397.sumstats.gz")

BMI_16_53797565_53848561=fread("../ukb_imp_v3.UKB_caucasians_BMIwaisthip_AsthmaAndT2D_INT-BMI_withagesex_041720.BoltLMM.snp_stats.all_chr_16_53797565_53848561.sumstats.gz")
##############
BMI_colnames=colnames(BMI_16_53797565_53848561)
colnames(BMI_16_11031741_11307024)=BMI_colnames
colnames(BMI_16_11034728_11450487)=BMI_colnames
colnames(BMI_16_11031741_11414898)=BMI_colnames
colnames(BMI_16_27202702_27371397)=BMI_colnames
final_BMI_colnemas=str_c("BMI_",BMI_colnames)
BMI=rbind(BMI_16_11031741_11307024,BMI_16_11031741_11414898,BMI_16_11034728_11450487,BMI_16_27202702_27371397,BMI_16_53797565_53848561)

colnames(BMI)=final_BMI_colnemas
head(BMI)
```

# Step1: Prepare the data
```{r,echo=T}
# for merge two data traits and finding the common SNPs
data <- gwas_merge(asthma, BMI, snp_name_cols = c("asthma_SNP", "BMI_SNP"), 
                       beta_hat_cols = c("asthma_BETA", "BMI_BETA"), 
                       se_cols = c("asthma_SE", "BMI_SE"), 
                       A1_cols = c("asthma_REF", "BMI_REF"), 
                       A2_cols = c("asthma_ALT", "BMI_ALT"))

# formated data
head(data)

dim(data)
```

Warning: Among the two traits, only 1081 common SNP variants, small number will lead to poor estimates of the mixing parameters whih leads to bad model comparisons 

# Step 2: Analysis-calculate the nuisance parameter
```{r}
set.seed(100)
varlist <- data$snp
params <- est_cause_params(data, varlist)
params

head(params$mix_grid)
```

We have estimated that 16% of variants have trait 1 variance and trait 2 equal to 0 meaning that they have no association with either trait.

# Step 3:LD Pruning 

This step requires LD estimates. Download estimates of chromsome 16 made in the 1000 Genomes CEU population from "https://zenodo.org/record/1464357#.W8a-fxROmV4"

```{r}
#download.file("https://zenodo.org/record/1464357/files/chr16_AF0.05_0.1.RDS?download=1", destfile = "../chr16_AF0.05_0.1.RDS")
#download.file("https://zenodo.org/record/1464357/files/chr16_AF0.05_snpdata.RDS?download=1", destfile="../chr16_AF0.05_snpdata.RDS")
```


```{r}
ld <- readRDS("../chr16_AF0.05_0.1.RDS")
snp_info <- readRDS("../chr16_AF0.05_snpdata.RDS")

head(ld)
head(snp_info)
```

The snp_info data frame contains information about all of the chromosome #16 variants with allele frequency greater than 0.05. The only piece of information we need from this data frame is the list of variants SNP column in snp_info data set which provides the total list of variants used in the LD calculations.
 
```{r}
variants <- data %>% mutate(pval1 = 2*pnorm(abs(beta_hat_1/seb1), lower.tail=FALSE))
pruned <- ld_prune(variants = variants, 
                            ld = ld, total_ld_variants = snp_info$SNP, 
                            pval_cols = c("pval1"), 
                            pval_thresh = c(1e-3))

length(pruned)
```
 
# Step4: Fit CAUSE
```{r}
res <- cause(X=data, variants = pruned, param_ests = params)
```

# Step 5: Results
```{r}
names(res)
res$elpd

```
model1, model2: The models being compared

delta_elpd: Estimated difference in elpd. If delta_elpd is negative, model 2 is a better fit
Here it seems model1 is better
se_delta_elpd: Estimated standard error of delta_elpd

z: delta_elpd/se_delta_elpd. A z-score that can be compared to a normal distribution to test if the difference in model fit is significant.



```{r}

summary(res, ci_size=0.95)
```



```{r}
plot(res$sharing)
plot(res, type="data")
```

